sdk/trace/build.gradle.kts

```diff
@@ -22,7 +22,7 @@ dependencies {
   api(project(":api:all"))
   api(project(":sdk:common"))
 
-  compileOnly(project(":api:incubator"))
+  optionalCompileOnly(project(":api:incubator"))
   compileOnly(project(":sdk:trace-shaded-deps"))
```

Comment:
If we go in this direction, we'll need to decide when its appropriate for a dependency to be `compileOnly` vs. `optionalCompileOnly`.

Originally, I just updated gradle maven publishing to include all `compileOnly` dependencies as optional dependencies in pom.xml, but that resulted in [these standard compileOnly annotation dependencies](https://github.com/open-telemetry/opentelemetry-java/blob/main/buildSrc/src/main/kotlin/otel.java-conventions.gradle.kts#L246-L248) being included in every pom.xml, which doesn't seem right.